axelor-base/src/main/java/com/axelor/apps/base/service/pricing/PricingComputer.java

```diff
@@ -197,24 +200,50 @@ protected void computeResultFormulaAndApply(Pricing pricing, PricingLine pricing
     resultPricingRuleList.add(pricing.getResult3PricingRule());
     resultPricingRuleList.add(pricing.getResult4PricingRule());
 
-    resultPricingRuleList.stream()
-        .filter(Objects::nonNull)
-        .forEach(
-            resultPricingRule -> {
-              MetaField fieldToPopulate = resultPricingRule.getFieldToPopulate();
-              Object result = scriptHelper.eval(resultPricingRule.getFormula());
-              if (fieldToPopulate != null) {
-                Mapper.of(classModel).set(model, fieldToPopulate.getName(), result);
-              }
-              if (!StringUtils.isBlank(resultPricingRule.getTempVarName())) {
-                LOG.debug(
-                    "Adding result temp variable {} in context",
-                    resultPricingRule.getTempVarName());
-                putInContext(resultPricingRule.getTempVarName(), result);
-              }
-              notifyResultPricingRule(resultPricingRule, result);
-              notifyFieldToPopulate(fieldToPopulate);
-            });
+    for (PricingRule resultPricingRule : resultPricingRuleList) {
+      if (resultPricingRule != null) {
```

Comment:
I'm surprised we could have a list with null element here. I don't know if it can happen but we should always avoid adding null element in a list